lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx

```diff
@@ -31,384 +19,321 @@ describe('NxSegmentedButton', function() {
+  it('calls onToggleOpen if a click happens anywhere when the dropdown is already open', async function() {
+    const user = userEvent.setup(),
+        onToggleOpen = jest.fn();
 
+    quickRender({ onToggleOpen, isOpen: true });
     expect(onToggleOpen).not.toHaveBeenCalled();
 
-    act(() => {
-      document.dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-
-    component!.update();
-
-    expect(onToggleOpen).toHaveBeenCalled();
+    await user.click(document.body);
+    expect(onToggleOpen).toHaveBeenCalledTimes(1);
   });
 
-  it('does not call onToggleOpen if a click happens anywhere when the dropdown is closed', function() {
-    const onToggleOpen = jest.fn(),
-        component = getMounted({ onToggleOpen }, { attachTo: container });
+  it('does not call onToggleOpen if a click happens anywhere aside from the'
+    + 'toggle button when the dropdown is closed', async function() {
+    const user = userEvent.setup(),
+        onToggleOpen = jest.fn();
 
+    quickRender({ onToggleOpen });
     expect(onToggleOpen).not.toHaveBeenCalled();
 
-    act(() => {
-      document.dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(document.body);
     expect(onToggleOpen).not.toHaveBeenCalled();
   });
 
-  it('does not call onToggleOpen if a click happens anywhere when the dropdown is disabled', function() {
-    const onToggleOpen = jest.fn(),
-        component = getMounted({ onToggleOpen, isOpen: true, disabled: true }, { attachTo: container });
+  it('does not call onToggleOpen if a click happens anywhere when the element is disabled', async function() {
+    const user = userEvent.setup(),
+        onToggleOpen = jest.fn();
 
+    quickRender({ onToggleOpen, isOpen: true, disabled: true });
     expect(onToggleOpen).not.toHaveBeenCalled();
 
-    act(() => {
-      document.dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(document.body);
     expect(onToggleOpen).not.toHaveBeenCalled();
   });
 
-  it('calls onToggleOpen if ESC is pressed within the component while the dropdown is open', function() {
-    const onToggleOpen = jest.fn(),
-        component = getMounted({ onToggleOpen, isOpen: true });
+  it('calls onToggleOpen if ESC is pressed within the component while the dropdown is open', async function() {
+    const user = userEvent.setup(),
+        onToggleOpen = jest.fn(),
+        component = renderEl({ onToggleOpen, isOpen: true })!,
+        dropdownBtn = within(component).getByRole('button', { name: 'more options' });
 
-    component.simulate('keyDown', { key: 'Escape', preventDefault: jest.fn() });
+    dropdownBtn.focus();
+    await user.keyboard('{Escape}');
     expect(onToggleOpen).toHaveBeenCalled();
   });
 
   it('calls preventDefault on Escape keydown', function() {
```

Comment:
See my recent commits in the `NxFilterDropdown` branch for an elaboration on this.  It is only supposed to call `preventDefault` when the dropdown is open.

Reply:
> See my recent commits in the `NxFilterDropdown` branch for an elaboration on this. It is only supposed to call `preventDefault` when the dropdown is open.

Thanks for pointing that out- updated here: [f25e053](https://github.com/sonatype/sonatype-react-shared-components/pull/927/commits/f25e053572e0e0069e299e9dbef87382cb8484b0)